squad-server/index.js

```diff
@@ -252,6 +252,48 @@ export default class SquadServer extends EventEmitter {
     await this.logParser.watch();
   }
 
+  async setupAdminList(remoteAdminLists) {
+    try {
+      const remoteAdmins = {
+        admins: {},
+        whitelist: {},
+        groups: {}
+      };
+
+      for (let idx = 0; idx < remoteAdminLists.length; idx++) {
+        const list = remoteAdminLists[idx];
+
+        const resp = await axios({
+          method: 'GET',
+          url: `${list}`
+        });
+
+        const rawData = resp.data;
+
+        const groupRgx = /(?<=Group=)(.*?):(.*)(?=\n)/g;
+        const adminRgx = /(?<=Admin=)(\d+):(\S+)(?=\s)/g;
+
+        /* eslint-disable no-unused-vars */
+        for (const [match, groupID, groupPerms] of rawData.matchAll(groupRgx)) {
+          remoteAdmins.groups[`${idx}-${groupID}`] = groupPerms.split(',');
+        }
+        for (const [match, steamID, groupID] of rawData.matchAll(adminRgx)) {
+          const perms = remoteAdmins.groups[`${idx}-${groupID}`];
+          if (!(perms.includes('reserve') && perms.length === 1)) {
+            remoteAdmins.admins[steamID] = `${idx}-${groupID}`;
```

Comment:
I'd suggest just a single list/set to store the Steam ID with the perms, e.g.:
```
[
  {
    steamID: "xxx",
    permA: true.
    permB: true
    // etc.
  }
]
```

Then add methods to filter this, e.g. `getAdminInfoBySteamID(steamID)` or `getAdminsWithPermission(permA)`, similar to the player list.

Reply:
I have done essentially this but instead of a list it is a dictionary for easy lookups of steamID's